--- a/configure.ac
+++ b/configure.ac
@@ -1080,6 +1080,14 @@
         darwin
 #elif defined(__VXWORKS__)
         vxworks
+#elif defined(__MSYS__)
+# if defined(__x86_64__)
+        x86_64-msys
+# elif defined(__i386__)
+        i386-msys
+# else
+#   error unknown platform triplet
+# endif
 #elif defined(__CYGWIN__)
 # if defined(__x86_64__)
         x86_64-cygwin
@@ -1510,7 +1526,7 @@
   case $ac_sys_system in
     CYGWIN*)
           LDLIBRARY='libpython$(LDVERSION).dll.a'
-          DLLLIBRARY='libpython$(LDVERSION).dll'
+          DLLLIBRARY='msys-python$(LDVERSION).dll'
           ;;
     SunOS*)
 	  LDLIBRARY='libpython$(LDVERSION).so'
--- a/Lib/ctypes/__init__.py
+++ b/Lib/ctypes/__init__.py
@@ -467,7 +467,7 @@
 if _os.name == "nt":
     pythonapi = PyDLL("python dll", None, _sys.dllhandle)
 elif _sys.platform == "cygwin":
-    pythonapi = PyDLL("libpython%d.%d.dll" % _sys.version_info[:2])
+    pythonapi = PyDLL("msys-python%d.%d.dll" % _sys.version_info[:2])
 else:
     pythonapi = PyDLL(None)
 
--- a/Lib/ctypes/util.py
+++ b/Lib/ctypes/util.py
@@ -390,8 +390,8 @@
             print(f"crypto\t:: {find_library('crypto')}")
             print(f"crypto\t:: {cdll.LoadLibrary(find_library('crypto'))}")
         elif sys.platform == "cygwin":
-            print(cdll.LoadLibrary("cygbz2-1.dll"))
-            print(cdll.LoadLibrary("cygcrypt-0.dll"))
+            print(cdll.LoadLibrary("msys-bz2-1.dll"))
+            print(cdll.LoadLibrary("msys-crypt-0.dll"))
             print(find_library("crypt"))
         else:
             print(cdll.LoadLibrary("libm.so"))
--- a/Lib/sysconfig.py
+++ b/Lib/sysconfig.py
@@ -812,6 +812,13 @@
         m = rel_re.match(release)
         if m:
             release = m.group()
+    elif osname[:4] == "msys":
+        osname = "msys"
+        import re
+        rel_re = re.compile (r'[\d.]+')
+        m = rel_re.match(release)
+        if m:
+            release = m.group()
     elif osname[:6] == "darwin":
         import _osx_support
         osname, release, machine = _osx_support.get_platform_osx(
